File: series3_handout/1d-FD/plot_convergence.py
from numpy import *
from pylab import *
errors = loadtxt("errors_fd.txt")
resolutions = loadtxt("resolutions_fd.txt")
rate = polyfit(log(resolutions), log(errors), 1)[0]
print("Approximate order: %1.2f" %
      round(rate, 2))

loglog(resolutions, errors, '-o', label="$||u_N-u||_\\infty$")
loglog(resolutions, errors[-1]*resolutions**rate/(resolutions[-1]**rate), '--', label='$N^{%0.2f}$' % rate)
legend()
ylabel("errors")
xlabel("$N$")
grid("on")
show()

File: series3_handout/1d-FD/plot_convergence.m
errors = load("errors_fd.txt");
resolutions = load("resolutions_fd.txt");
rate = polyfit(log(resolutions), log(errors), 1);
rate = rate(1);
disp("Approximate order: ");
disp(round(rate, 2));
N = length(errors);

loglog(resolutions, errors, '-o')
hold on
loglog(resolutions, errors(N-1)*resolutions.^rate/(resolutions(N-1).^rate), '--')
xlabel('# cells')
ylabel('L^1 error')
axis([resolutions(1) resolutions(N) -inf inf]);
label = strcat('N^{', num2str(rate,3), '}');
legend('||u_N-u||_\infty', label, 'Location', 'northeast')
grid on
